src/sage/combinat/specht_module.py

```diff
@@ -18,15 +25,227 @@
 # ****************************************************************************
 
 from sage.misc.cachefunc import cached_method
+from sage.misc.lazy_attribute import lazy_attribute
 from sage.combinat.diagram import Diagram
+from sage.combinat.partition import _Partitions
+from sage.combinat.free_module import CombinatorialFreeModule
 from sage.sets.family import Family
 from sage.matrix.constructor import matrix
 from sage.rings.rational_field import QQ
-from sage.modules.with_basis.subquotient import SubmoduleWithBasis
+from sage.modules.with_basis.subquotient import SubmoduleWithBasis, QuotientModuleWithBasis
+from sage.modules.free_module_element import vector
 from sage.categories.modules_with_basis import ModulesWithBasis
 
+class SymmetricGroupRepresentation:
+    """
+    Mixin class for symmetric group (algebra) representations.
+    """
+    def __init__(self, SGA):
+        """
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: SM = Partition([3,1,1]).specht_module(GF(3))
+            sage: TestSuite(SM).run()
+        """
+        self._semigroup = SGA.group()
+        self._SGA = SGA
+
+    @cached_method
+    def frobenius_image(self):
+        r"""
+        Return the Frobenius image of ``self``.
+
+        The Frobenius map is defined as the map to symmetric functions
+
+        .. MATH::
+
+            F(\chi) = \frac{1}{n!} \sum_{w \in S_n} \chi(w) p_{\rho(w)},
+
+        where `\chi` is the character of the `S_n`-module ``self``,
+        `p_{\lambda}` is the powersum symmetric function basis element
+        indexed by `\lambda`, and `\rho(w)` is partition of the cycle type
+        of `w`. Specifically, this map takes irreducible representations
```

Comment:
By "and `\rho(w)` is partition of the cycle type of `w`", do you mean ``... is the cycle type of `w` as a partition?"